Content.Server/Singularity/EntitySystems/SingularityGeneratorSystem.cs

```diff
@@ -42,6 +47,12 @@ private void OnPassThreshold(EntityUid uid, SingularityGeneratorComponent? comp)
     {
         if (!Resolve(uid, ref comp))
             return;
+        if (_lock.IsLocked(uid))
+        {
+            _popup.PopupEntity(Loc.GetString("singularity-generator-component-locked"), uid);
+            SetPower(uid, 0, comp);
+            return;
+        }
 
```

Comment:
Hugboxing?

Reply:
I mean the main idea is to try to lower the amount of both raider loosing by adding more friction to the process, and also engi loosing roundstart by trying to get someone vaguely experienced to okay a generator setup, since immediate evacs due to looses just aren't particularly fun and are for all intents and purposes unrecoverable, which can lead to feelsbads for a lot of people (e.g. nukies loosing their roll, or honestly any actual antag)